src/models/ValidationTypes.js

```diff
@@ -0,0 +1,121 @@
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import { translate as t } from '@nextcloud/l10n'
+
+import IconEMail from 'vue-material-design-icons/Email.vue'
+import IconPhone from 'vue-material-design-icons/Phone.vue'
+import IconRegex from 'vue-material-design-icons/Regex.vue'
+import IconTextShort from 'vue-material-design-icons/TextShort.vue'
+import IconNumeric from 'vue-material-design-icons/Numeric.vue'
+
+/**
+ * @callback ValidationFunction
+ * @param {string} input User input text
+ * @param {?Record<any>} options Optional setting for validation, like regex pattern.
+ * @return {boolean} True if the input is valid, false otherwise
+ */
+/**
+ * @typedef {object} ValidationType
+ * @property {string} label The validation-type label, that users will see.
+ * @property {string} inputType The HTML <input> type used.
+ * @property {string} errorMessage The error message shown if the validation fails.
+ * @property {string|undefined} createPlaceholder *optional* A typed placeholder that is visible in edit-mode, to indicate a submission form-input field
+ * @property {string|undefined} submitPlaceholder *optional* A typed placeholder that is visible in submit-mode, to indicate a form input-field
+ * @property {import('vue').Component} icon The icon users will see on the input field.
+ * @property {ValidationFunction} validate Function for validating user input to match the selected input type.
+ */
+
+// !! Keep in SYNC with lib/Constants.php for supported types of input validation !!
+export default {
+	/**
+	 * Default, not validated, text input
+	 *
+	 * @type {ValidationType}
+	 */
+	text: {
+		icon: IconTextShort,
+		inputType: 'text',
+		label: t('forms', 'Text'),
+		validate: () => true,
+		errorMessage: '',
+	},
+
+	/**
+	 * Phone number validation
+	 *
+	 * @type {ValidationType}
+	 */
+	phone: {
+		icon: IconPhone,
+		inputType: 'tel',
+		label: t('forms', 'Phone number'),
+		// Remove common separator symbols, like space or braces, and validate rest are pure numbers
+		validate: (input) => /^\+?[0-9]{3,}$/.test(input.replace(/[\s()-/x.]/ig, '')),
+		errorMessage: t('forms', 'The input is not a valid phone number'),
+		createPlaceholder: t('forms', 'People can enter a telephone number'),
+		submitPlaceholder: t('forms', 'Enter a telephone number'),
+	},
+
+	/**
+	 * Email address validation
+	 *
+	 * @type {ValidationType}
+	 */
+	email: {
+		icon: IconEMail,
+		inputType: 'email',
+		label: t('forms', 'Email address'),
+		validate: (input) => /^\w+([.-]?\w+)*@\w+([.-]?\w+)*(\.\w{2,3})+$/.test(input),
```

Comment:
With this validation pattern we're not matching all valid mail addresses... For example + is not allowed in your validation, but in RFC

```suggestion
		validate: (input) => /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,24}))$/.test(input),
```
I got this one from https://codepen.io/CSWApps/pen/MmpBjV

However, most sites suggest to use an API for address validation as the complexity needed for a complete regex validation hits performance. But I'm not sure if we want to leave NC for another service here.

Reply:
Also some more input on this from Mozilla: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/email#validation

Reply:
Ok I would vote for making the frontend code simpler. On the backend we already use the Nextcloud core service to validate the email.

So on frontend we should just make sure it somehow looks like an email. how about:
`[^@]+@[^@]+`
This one will only forbid one case: quoted `@` symobol (as something like `"foo@bar"@example.com` is valid.